qt/python/mantidqtinterfaces/mantidqtinterfaces/DGSPlanner/DGSPlannerGUI.py

```diff
@@ -159,30 +159,14 @@ def eiWavelengthUpdateEvent(self):
             self.dimensionWidget.set_editMax4(upperBound)
 
     def instrumentUpdateEvent(self):
-        if self.masterDict['instrument'] == 'WAND\u00B2':
-            self.instrumentWAND = True
-            # change the ui accordingly
+        changeToElastic = True if self.masterDict['instrument'] in ['WAND\u00B2'] else False
+        if changeToElastic:
             self.dimensionWidget.toggleDeltaE(False)
-            self.instrumentWidget.setLabelEi('Input Wavelength')
-            self.instrumentWidget.setEiVal(str(1.488))
-
-            self.instrumentWidget.setGoniometerNames(['s1', 'sgl', 'sgu'])
-            self.instrumentWidget.setGoniometerDirections(['0,1,0', '1,0,0', '0,0,1'])
-            self.instrumentWidget.setGoniometerRotationSense([1, -1, -1])
-            self.instrumentWidget.updateAll()
-
-            self.eiWavelengthUpdateEvent()
-        else:
-            if self.instrumentWAND:
-                self.instrumentWAND = False
-                self.dimensionWidget.toggleDeltaE(True)
-                self.instrumentWidget.setLabelEi('Incident Energy')
-                self.instrumentWidget.setEiVal(str(10.0))
-
-                self.instrumentWidget.setGoniometerNames(['psi', 'gl', 'gs'])
-                self.instrumentWidget.setGoniometerDirections(['0,1,0', '0,0,1', '1,0,0'])
-                self.instrumentWidget.setGoniometerRotationSense([1, 1, 1])
-                self.instrumentWidget.updateAll()
+            self.instrumentElastic = True
+        elif self.instrumentElastic:
+            self.dimensionWidget.toggleDeltaE(True)
+            self.instrumentElastic = False
```

Comment:
all this if else is unnecessary. Try something like:
```
self.instrumentElastic = changeToElastic
self.dimensionWidget.toggleDeltaE(not changeToElastic)
```

Reply:
I incorporated this suggestion. But I added the conditional so that the DeltaE is not always cleared out when switching between inelastic instruments. Only when switching between inelastic/elastic instrument, should the toggleDeltaE be activated. Try it with ARCS and SEQ by first typing in values for deltaE.